apps/omg/lib/omg/state/core.ex

```diff
@@ -168,11 +166,11 @@ defmodule OMG.State.Core do
     tx_hash = Transaction.raw_txhash(tx)
     outputs = Transaction.get_outputs(tx)
 
-    with :ok <- validate_block_size(state),
-         {:ok, input_amounts_by_currency} <- correct_inputs?(state, tx),
-         output_amounts_by_currency = get_amounts_by_currency(outputs),
-         :ok <- amounts_add_up?(input_amounts_by_currency, output_amounts_by_currency),
-         :ok <- transaction_covers_fee?(input_amounts_by_currency, output_amounts_by_currency, fees) do
+    with :ok <- Validator.validate_block_size(state),
+         {:ok, input_amounts_by_currency} <- Validator.correct_inputs?(state, tx),
+         output_amounts_by_currency = Validator.get_amounts_by_currency(outputs),
+         :ok <- Validator.amounts_add_up?(input_amounts_by_currency, output_amounts_by_currency),
+         :ok <- Validator.transaction_covers_fee?(input_amounts_by_currency, output_amounts_by_currency, fees) do
```

Comment:
Nice move, then this module isn't interested with all details of validation. How about of wrapping all above into `can_apply_spend?(state, tx)` function :question:

Reply:
yeah, make sense!